homeassistant/components/luxtronik/sensor.py

```diff
@@ -0,0 +1,613 @@
+            sensor_id = sensor_cfg[CONF_ID]
+            if "." in sensor_id:
+                group = sensor_id.split(".")[0]
+                sensor_id = sensor_id.split(".")[1]
+            else:
+                group = sensor_cfg[CONF_GROUP]
+            sensor = luxtronik.get_sensor(group, sensor_id)
+            if sensor:
+                name = (
+                    sensor.name
+                    if not sensor_cfg.get(CONF_FRIENDLY_NAME)
+                    else sensor_cfg.get(CONF_FRIENDLY_NAME)
+                )
+                icon = (
+                    ICONS.get(sensor.measurement_type)
+                    if not sensor_cfg.get(CONF_ICON)
+                    else sensor_cfg.get(CONF_ICON)
+                )
+                entities += [
+                    LuxtronikSensor(
+                        hass=hass,
+                        luxtronik=luxtronik,
+                        deviceInfo=device_info,
+                        sensor_key=f"{group}.{sensor_id}",
+                        unique_id=sensor_id,
+                        name=name,
+                        icon=icon,
+                        device_class=DEVICE_CLASSES.get(
+                            sensor.measurement_type, DEFAULT_DEVICE_CLASS
+                        ),
+                        state_class=None,
+                        unit_of_measurement=UNITS.get(sensor.measurement_type),
+                    )
+                ]
+            else:
+                LOGGER.warning(
+                    "%s.sensor.async_setup_platform: Invalid Luxtronik ID %s in group %s",
+                    DOMAIN,
+                    sensor_id,
+                    group,
+                )
+        # endregion yaml sensors part
+
+    async_add_entities(entities)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Luxtronik sensor from ConfigEntry."""
+    luxtronik: LuxtronikDevice = hass.data.get(DOMAIN)
+    if not luxtronik:
+        LOGGER.warning("%s.sensor.async_setup_entry no luxtronik!", DOMAIN)
+        return False
+
+    device_info = hass.data[f"{DOMAIN}_DeviceInfo"]
+
+    # Build Sensor names with local language:
+    lang = config_entry.options.get(CONF_LANGUAGE_SENSOR_NAMES)
+    hass.data[f"{DOMAIN}_language"] = lang
+    text_time = get_sensor_text(lang, "time")
+    text_temp = get_sensor_text(lang, "temperature")
+    text_heat_source_output = get_sensor_text(lang, "heat_source_output")
+    text_heat_source_input = get_sensor_text(lang, "heat_source_input")
+    text_outdoor = get_sensor_text(lang, "outdoor")
+    text_average = get_sensor_text(lang, "average")
+    text_compressor_impulses = get_sensor_text(lang, "compressor_impulses")
+    text_operation_hours = get_sensor_text(lang, "operation_hours")
+    text_heat_amount_counter = get_sensor_text(lang, "heat_amount_counter")
+    entities = [
+        LuxtronikStatusSensor(
```

Comment:
This integration would benefit from using `SensorEntityDescriptions` instead of passing in all the info.